src/operators/Remap/avtRemapFilter.C

```diff
@@ -131,447 +138,476 @@ avtRemapFilter::SetAtts(const AttributeGroup *a)
+                clipperTop->SetInputConnection(last->GetOutputPort());
+                last = clipperTop;
+
+                // Bottom plane
+                clipperBottom->SetInputData(in_ds);
+                clipperBottom->SetClipFunction(funcsArrayY[j+1]);
+                clipperBottom->SetInsideOut(true);
+                clipperBottom->SetRemoveWholeCells(false);
+                clipperBottom->SetInputConnection(last->GetOutputPort());
+                last = clipperBottom;
+
+                if (is3D)
                 {
-                    value += myVariable->GetComponent(tuple, 0) *
-                        subCellVolumes->GetComponent(tuple, 0);
+                    // Front plane
+                    clipperFront->SetInputData(in_ds);
+                    clipperFront->SetClipFunction(funcsArrayZ[k]);
+                    clipperFront->SetInsideOut(false);
+                    clipperFront->SetRemoveWholeCells(false);
+                    clipperFront->SetInputConnection(last->GetOutputPort());
+                    last = clipperFront;
+
+                    // Back plane
+                    clipperBack->SetInputData(in_ds);
+                    clipperBack->SetClipFunction(funcsArrayZ[k+1]);
+                    clipperBack->SetInsideOut(true);
+                    clipperBack->SetRemoveWholeCells(false);
+                    clipperBack->SetInputConnection(last->GetOutputPort());
+                    last = clipperBack;
                 }
-                value /= rCellVolume;
-                vars[vdx]->SetComponent(rCell, 0, value);
-            }
-            else if (varType == RemapAttributes::extrinsic) // like mass
-            {
-                for (vtkIdType tuple = 0;
-                     tuple < myVariable->GetNumberOfTuples(); tuple++)
+
+                last->Update();
+                vtkUnstructuredGrid* ug = last->GetOutput();
+                debug5 << "Cell " << rCell << " has been clipped." << std::endl;
+                // Done clipping
+        
+                // Collection of clipped cells from the original grid that now take the shape
+                // of the rgrid cell, could be something like this:
+                // + -------------- +
+                // |   |      /     |
+                // | 1 | 3   /  5   |
+                // |   ^    /------ |
+                // |  / \  /    |   |
+                // | /   \/  4  | 6 |
+                // |/  2  \     |   |
+                // + -------------- +
+    
+                // --- Calculate volume of subcells --- //
+                // Now that we have the unstrucutred grid from the clipping, we can loop
+                // over the subcells in that grid and calculate the volumes. We also need
+                // the ratio between old and new volumes to weigh the variables (like mass,
+                // density, pressure, etc.).
+
+                vtkDoubleArray* subCellVolumes = CalculateCellVolumes(ug, "subCellVolume");
```

Comment:
Please delete the array allocated by CalculateCellVolumes when you are finished with it.

Reply:
Done